### ISSUE TITLE
NIPS 23 | DPOK: Reinforcement Learning for Fine-tuning Text-to-Image Diffusion Models

### ISSUE THREAD

**Issue by hiahei** [AI-Infra-Team/awesome-papers#170]
# DPOK: Reinforcement Learning for Fine-tuning Text-to-Image Diffusion Models
Authors:                                                                                     
Ying Fan, Olivia Watkins, Yuqing Du, Hao Liu, Moonkyung Ryu, Craig Boutilier, Pieter Abbeel, Mohammad Ghavamzadeh, Kangwook Lee, Kimin Lee
[Paper](https://arxiv.org/abs/2305.16381) | [code](https://github.com/google-research/google-research/tree/master/dpok)

## Motivation
How to utilize **Reinforcement Learning (RL)** to fine-tune the Text-to-Image diffusion model, so as to better align with practical application (such as human preferences, image quality, or specific metrics).

There are four three contributions:
- Present DPOK: Diffusion Policy Optimization with KL regularization, which utilizes KL regularization w.r.t. the pre-trained T2I model as an implicit reward to stabilize RL fine-tuning and frame the optimization of the expected reward (w.r.t. an LHF-reward) of the images generated by a diffusion model given text prompts as an online RL problem.
- Study incorporating KL regularization into supervised fine-tuning of diffusion models, which can mitigate some failure modes. This also allows a fairer comparison with our RL technique.
- Discuss the key differences between supervised fine-tuning and online fine-tuning of T2I models.

## Fine-tuning of Diffusion Models
We consider the use of denoising diffusion probabilistic models (**DDPMs**) for image generation and draw our notation and problem formulation from.

<div align="center">
<img width="961" height="158" alt="Image" src="https://github.com/user-attachments/assets/c852ace6-c4be-47b8-a3f8-8a51febcd7a7" />
</div>

It can be seen from the MDP formulation in (4) that the system starts by sampling its initial state $s_0$ from the Gaussian distribution $N(0, I)$, similar to the first state of the dinoising process $x_T$ .

write the gradient of the objective as:
<div align="center">
<img width="866" height="86" alt="Image" src="https://github.com/user-attachments/assets/448cf266-8df5-4cd3-a45a-07894f1a3874" />
</div>

where $r(x_0, z)$ is the reward.

Then introduce KL regularization into the objective to constraint parameter update:
<div align="center">
<img width="954" height="92" alt="Image" src="https://github.com/user-attachments/assets/50a24338-4dca-4ff9-afa4-01e6ec9a55a1" />
</div>

Another task of DPOK is to apply KL regularization to the SFT training process, and also to update the objective function of SFT.

## Experiment

<div align="center">
<img width="1453" height="390" alt="Image" src="https://github.com/user-attachments/assets/88d10a2a-caba-47dd-9849-9f9a88142397" />
</div>

<div align="center">
<img width="1280" height="393" alt="Image" src="https://github.com/user-attachments/assets/ce5bde3c-1e45-4937-96aa-6bb7fbb65d86" />
</div>

<div align="center">
<img width="500" height="auto" alt="Image" src="https://github.com/user-attachments/assets/06e7083f-21ef-44f4-9583-b0d05b863c71" />
</div>

## Future Directions
- More efficient training with a broader range of diverse and complex prompts would be an interesting future direction to explore. (**Focus on reward design and sampling**)
- Exploring advanced policy gradient methods can be useful for further performance improvement. (**Focus on valid gradient and target function**)
